### PR TITLE
Adnuntius Bid Adaptor: handling deals better

### PR DESCRIPTION
**Pull request by antosarho**
## Type of change

- [x] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)

- [x] Does this change affect user-facing APIs or examples documented on http://prebid.org?

## Description of change
This change adds better support for deals to the Adnuntius Bid Adaptor.

It adds a new maxDeals parameter, which defaults to 0, that allows the adaptor to return responses for both the highest bidding advertisement AND (when maxDeals > 0) deal advertisements.

## Other information

This change introduces a new optional parameter, which will require an update to the adaptor documentation but it does not affect functionality for existing users.

### CODE REVIEW

**Review by osazos — COMMENTED** [review]
The [prebid.github.io](https://github.com/prebid/prebid.github.io/pulls) PR is missing, it would be nice to open it in the same time.

By the way, LGTM for the code review.

File: modules/adnuntiusBidAdapter.js [hunk 1]
```diff
@@ -71,12 +51,23 @@ const getUsi = function (meta, ortb2, bidderRequest) {
...
+    BIDDER_CODE_DEAL_ALIAS_3,
+    BIDDER_CODE_DEAL_ALIAS_4,
+    BIDDER_CODE_DEAL_ALIAS_5
+  ],
   gvlid: GVLID,
   supportedMediaTypes: [BANNER, VIDEO],
   isBidRequestValid: function (bid) {
-    return !!(bid.bidId || (bid.params.member && bid.params.invCode));
+    // The auId MUST be a hexadecimal string
+    const validAuId = bid.params.auId && bid.params.auId.length > 0 && AU_ID_REGEX.test(bid.params.auId);
```
[the remark this answers is not in the document]
Comment (antosarho) [reply modules/adnuntiusBidAdapter.js]: Good suggestion. Done.

**Comment by antosarho** [comment]
> The [prebid.github.io](https://github.com/prebid/prebid.github.io/pulls) PR is missing, it would be nice to open it in the same time.
> 
> By the way, LGTM for the code review.

And the doco update is here (done by a co-worker): https://github.com/prebid/prebid.github.io/pull/4591

**Review by osazos — APPROVED** [review]

**Comment by robertrmartinez** [comment]
Hello @antosarho

I work for Magnite and while testing a newer version of prebid I noticed that one of our pubs who uses an alias of your adapter had their spend through it drop to zero.

This is the only PR for your adapter that is between the two versions in question (7.50.0 and 7.54.0)

I just wanted to reach out and confirm if this is expected on your side?

I see the code Here: https://github.com/Adnuntius/Prebid.js/blob/3b7c8d62318d71bc95431fbaaffbb4765f36cc88/modules/adnuntiusBidAdapter.js#L179-L188

which seems to want to filter out any alias bidders.

Is there something the publisher needs to change, such as this new param `maxDeals` which defaults to zero now? Which is not something sent previously?

Or is this a known thing and alias bidders are no longer able to go through adnuntius?

Thanks!

**Comment by antosarho** [comment]
Hello @robertrmartinez 

I must apologise for the regression sneaking in and I thank you for being so thorough to pinpoint exactly where the problem lies.

I've now created this pull request that will address the issue. 

Unfortunately there is no fallback option to allow the use of aliases with the adnuntius adaptor in 7.54.0. 

Apologies once more, and thanks again.

Cheers!

**Comment by robertrmartinez** [comment]
Hey glad to help @antosarho 

One more question, one of our publishers uses an alias named `calt`

Will this still work? From looking at the PR it seams you only want aliases named `adndeal{1-5}`

So, is this something publishers need to migrate to? Understanding other alias names will not work?

Thanks!

**Comment by antosarho** [comment]
Hey @robertrmartinez,

The intention is to treat specific aliases in a special way because our ad server does something specific in these cases.

If the aliases are not `adndeal{1-5}`, it should work like normal -- that, at least, is the intention.

So if an alias is specified with `pbjs.aliasBidder('adnuntius', 'alt-adnuntius');`, then it should work fine.

Is that how magnite code specifies the alias? Or is there another way we should cater for?

Thanks again!

**Comment by robertrmartinez** [comment]
Hey @antosarho 

You are right, at a quick glance I misinterpreted the code change in #10232 

It is really saying

"If the bidder is NOT one of the deal aliases, add its bid response"

I thought it was the reverse!

Thanks for the quick fix!

**Pull request merged**